components/layouts/FullscreenLayout.tsx

```diff
@@ -8,13 +8,43 @@ import { useLocation } from "react-router";
 import { Toaster } from "react-hot-toast";
 import useResetPreline from "@/hooks/useResetPreline.ts";
 import { PostHogWrapperProvider } from "@/contexts/PostHogWrapperProvider.tsx";
+import { KEYRING_CHANGE_TIME } from "@/lib/keyring-manager.ts";
 
 export default function FullscreenLayout() {
   const navigation = useNavigation();
+  const navigate = useNavigate();
   const location = useLocation();
   const { isConnected } = useRpcClientStateful();
+  const [unlockTime] = useStorageState(
+    `local:${KEYRING_CHANGE_TIME}`,
+    Date.now(),
+  );
```

Comment:
I think we should use storage.watch, as I didn't see the full page recover after unlocking.

Reply:
Nvm, my bad, the process is right, I didn't reload the runtime.